src/libraries/Native/Unix/System.Native/pal_log.h

```diff
@@ -8,8 +8,12 @@
 
 PALEXPORT void SystemNative_Log(uint8_t* buffer, int32_t length);
 
+#if defined(TARGET_IOS) || defined(TARGET_TVOS)
+
 PALEXPORT int32_t SystemNative_InitializeTerminalAndSignalHandling(void);
 
 // Called by pal_signal.cpp to reinitialize the console on SIGCONT/SIGCHLD.
 void ReinitializeTerminal(void) {}
```

Comment:
Providing implementation in a header file like this will be a farm for multi-definition problems, etc. I would create a new file `pal_console_dummy.c`, move this into that file and include this file in the build for iOS and tvOS.

Once you do that, we should not need the other build system tweaks.

Reply:
Or: Add `HAS_CONSOLE_SIGNALS` define, define it in the build for !iOS && !tvOS, and ifdef out the calls to the console signal support in `pal_signal.c`.